### DISCUSSION TITLE
Airplay to AppleTV

### DISCUSSION THREAD

**Discussion by dunxd** [music-assistant/support#2528] [Feature requests and ideas]
### Current Status
Apple TV support has been added to 2.7.0 beta 11

### Original Post
[The docs say that AppleTV is not supported as an Airplay player due to the requirement for authentication](https://music-assistant.io/player-support/airplay/#known-issues-notes), but this may be fixed if there is demand. Adding this as a feature request because I would like it, and to gauge additional demand.

**Comment by adamskoreyko** [comment]
I would love this feature, the Apple TV integration with HA is very reliable and it would be outstanding to have them work in MA.

**Reply by marcohamersma** [reply]
Same, in my living room the Apple TV _is_ my smart speaker. Streaming to it with MA would be the dream

**Comment by saya6k** [comment]
I also want my Apple TV to integrate with Music Assistant!

**Reply by kdutton1964** [reply]
+1 I use Roon and HA with a quandary seeing as Roon needs password auth and HA cannot. It’s one or other for ATV use unless someone has alternate config to share. Checking to see progress in MA and this feature would be great option please and thank you. Computer Keith.

**Comment by ethan-tripp** [comment]
+1 for Apple TV with Music Assistant! Pretty much the main thing I was hoping to integrate and group with Airplay.

**Comment by zacburkemig** [comment]
Not having the Apple TV integration is the one thing keeping me from moving my home over to Music Assistant.

**Reply by ethan-tripp** [reply]
right there with you @zacburkemig

**Comment by kcofoni** [comment]
So nice Music Assistant Server! So good to use Apple TV as a main speaker in my living room. I am really impatient to be able to benefit from both. Thank you in advance and congratulations for this great job 🙂

**Comment by pgrill79** [comment]
Oh yes - really would love to see AppleTV supported. I have two usecases:
1) It allows me to stream to my Receiver which does not support AirPlay
2) It would allow me to integrate Hue Sync Box for light effects. The Hue Sync Box only can handle HDMI signals. So i could stream to my Apple TV and would have the light effects for the music

So would be happy to see this feature in Music Assistant (which is a great tool)

**Comment by mgoettelman** [comment]
Would love to see AppleTV supported!

**Comment by heccoder** [comment]
+1 for Apple TV support!

**Comment by icovelev** [comment]
The only thing current missing, which is for sure needed. A lot of people are using the Apple TV with receiver as a main music player.

**Comment by Pierrehagithub** [comment]
+1 for Apple TV support.
Thanks

**Comment by TheodorCZ1** [comment]
+1 for Apple TV PLEASEEE it would complete my music setup and allow to share music between multiple devices at once plsss...

**Comment by lewmilburn** [comment]
+1, would love this!

**Comment by drben1980** [comment]
+1 here, that would be amazing! I have 2 x apple TVs that I would love to be able to stream to. At the moment I am using airplay via an iPad but Music Assistant would be a much neater and more integrated solution. 

Thank you all so much for your work on Music Assistant, it's fantastic!

**Comment by ByteFetch** [comment]
+1 for Apple TV player support

**Comment by firstchair** [comment]
Big upvote for this :-D

**Comment by firstchair** [comment]
BTW could something like [pyatv](https://github.com/postlund/pyatv) not be used to make this happen?

**Reply by marcelveldt** [reply]
sure when time allows.

**Comment by marcelveldt** [comment]
Please stop with the +1 comments and if you want this feature just use the upvote button.
Nothing is going to make it faster by posting an additional comment. This will come when priority allows. And priority is chosen by the number of votes, not how loud people scream.

Thanks

**Reply by duckdude555** [reply]
Are there plans to implement this in an upcoming beta release? It looks like this is the top open feature request that hasn't been implemented yet. Thank you :)

**Reply by OzGav** [reply]
No there are no plans at this time. The project is swamped with existing tasks.

**Reply by marcelveldt** [reply]
This is currently being worked on but it will take some time to complete, we need to add the whole authentication logic for this to work

**Comment by dunxd** [comment]
I opened this issue in response to the text in [the Airplay documentation](https://www.music-assistant.io/player-support/airplay/):

> Because Apple TV's require authentication, they are not supported yet (but will be in the future if there's any demand)

I trust if this thread indicates a sufficient level of demand this may stay as a potential feature for the future. If not, I suggest editing the docs so they stop inviting the input.

**Reply by zacburkemig** [reply]
This is why I commented on this request as well. Came straight form the documentation to show my desire for the feature.

**Comment by heccoder** [comment]
If we look at the number of votes (based on the previous comment), this feature request ranks first among the discussion topics that have yet to be implemented, so hopefully this is reconsidered for implementation.

**Comment by normal-painter** [comment]
Please consider adding support for AppleTVs

**Comment by UgricMongol** [comment]
Please! Cannot live without it!

**Comment by joseluisalonso** [comment]
I Want  support for AppleTVs Please

**Comment by NieropT** [comment]
Would also love support!

**Comment by Andy071993** [comment]
+1 I would love if MA would have Apple TV support.

**Comment by YekaiZ** [comment]
Apple TV support pls!

**Comment by macgyver2089** [comment]
+1 for the appleTV support

**Comment by marcelveldt** [comment]
Apple TV is supported in the 2.7 beta. You need to go into player settings to do the one time pairing

**Reply by joseluisalonso** [reply]
Hello. thank you for your work 
 It is ok for apple TV Gen 4 (4k) ? 
where i can found the 2.a beta

**Reply by OzGav** [reply]
https://www.music-assistant.io/release/

**Reply by joseluisalonso** [reply]
Hello, I installed the version 2.7 beta. Great, we can use the Apple TV Gen 2.
 However, you have to redo the pairing after each restart of the Apple TV!!!
 Too bad.
 I hope there is a solution for this. 
Thanks in advance.

**Reply by OzGav** [reply]
I don't use Apple TV so can you explain why you have to restart it?

**Reply by joseluisalonso** [reply]
Hello. Because when I turn off the hi-fi, it also cuts off the Apple TV. 
And to avoid leaving the Apple TV on 24/7, 
the other devices paired with the Apple TV do not need to pair again after put off to on again

Additional information: Following the update to version 2.7.0b12,
 it was necessary to re-pair the Apple TV. 
It seems that Music Assistant does not retain the pairing information.

**Reply by OzGav** [reply]
Others using a 4K Gen 2 report pairing is not lost after a power cycle. Perhaps do some more testing including leaving it powered to see if you can provide any more information

**Reply by Hackashaq666** [reply]
After restarting the musicassistant container, I needed to re-pair my 3 Apple TVs.

**Reply by jozefKruszynski** [reply]
How are you guys running Music Assistant?
Addon or external docker?

**Reply by Hackashaq666** [reply]
I am running as docker container. Data volume is mapped to a directory outside of the container.

**Reply by joseluisalonso** [reply]
Hello.  
Thank you for your response. 
Surprisingly, the problem has disappeared!!  
That's great. Did you fix something in the latest beta?  
My Music Assistant is running on HA Yellow for information
Well, the important thing is that it works well. I will continue to test and keep you informed if there is any problem.  
On another note, can we discuss issues with DLNA? Problems with next, previous, correctly displaying the track duration, etc., etc.  
Thanks again for your work.

**Reply by Hackashaq666** [reply]
Ok, I restarted the music asssistant docker container twice in a row and all 3 apple tvs stayed paired.
2.7.0b13

**Reply by joseluisalonso** [reply]
Oops! Bad news.  
This morning on the 3 TV apple TV, the pairing disappeared.  
yesterday it was working fine. Nothing has been changed or done since yesterday. Only stopped the hi-fi and turned it on this morning!!!!  
Incomprehensible

**Reply by joseluisalonso** [reply]
I redid the pairing. Turned off the HiFi, then turned it on again. Pairing disappeared. Problem!

**Reply by OzGav** [reply]
These couple of sentence reports are not helpful. Provide clear details about what you are doing and EXACTLY what you are seeing and what you do to resolve it. We can’t recreate this and we can’t see what you can see so make it very clear.

**Reply by jozefKruszynski** [reply]
Also please give details regarding network setup, are you using vlans, enterprise network gear, mdns repeaters? The more info the better. Logs would also be very useful please.

In the music assistant interface go to Settings > Core, scroll to the bottom of the page and click the download server logfile button.

**Reply by duckdude555** [reply]
No pairing issues here despite multiple container and appleTv restarts. I had to enable encryption in the AirPlay settings to get audio to work. Thank you for implementing this feature!!!!

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
